### PR TITLE
feat: impl kcl config generation and add indent block for JSON/YAML inputs

### LINKED ISSUE #204: [Enhancement] Import Tool for the top level config keys with hyphens.
## Enhancement

For the following YAML

```yaml
a-b-c: a-b-c
```

The import tool should generate the following code

```python
{
    "a-b-c" = "a-b-c"
}
```

Instead of

```python
"a-b-c" = "a-b-c"
```

This is because the current version of KCL does not support the syntax mentioned in the above code.

### PR DESCRIPTION
**Pull request by Peefy**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix #204 

#### 2. What is the scope of this PR (e.g. component or file name):

+ pkg/tools/gen/genkcl_json.go
+ pkg/tools/gen/genkcl_yaml.go
+ pkg/tools/gen/template.go
+ pkg/tools/gen/testdata/json/**
+ pkg/tools/gen/testdata/yaml/**

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [x] Other

### CODE REVIEW

File: pkg/tools/gen/templates/kcl/config.gotmpl [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{{- if .Var }}{{ formatName .Var }}{{- if .IsUnion }}{{ " : " }}{{- else }}{{ " = " }}{{- end }}{{- end }}{{- if .Name }}{{ formatName .Name }}{{- end }}{{- if .Data }}{{- "{\n" }}
+{{- range .Data -}}
+{{- "    " }}{{ formatName .Key }}{{ " = " }}
+    {{- if isKclData .Value }}
```
Comment (jakezhu9) [inline pkg/tools/gen/templates/kcl/config.gotmpl]: duplicate with `data.gotmpl`. we can replace it with`{{- indentLines (include "data"  .) "    " }}`

Comment (Peefy) [reply pkg/tools/gen/templates/kcl/config.gotmpl]: > duplicate with `data.gotmpl`. we can replace it with`{{- indentLines (include "data" .) " " }}`

This is a good suggestion, but the four leading spaces of `}` and `]` have been added to hardcode in `data.tmpl`. Thus, the `config.tmpl` and `data.tmpl` should me merged in the future. 😄

Comment (jakezhu9) [reply pkg/tools/gen/templates/kcl/config.gotmpl]: Yes, we can merge them in the future. However I am a little confuse about what you meant by saying "the four leading spaces of } and ] have been added to hardcode in data.tmpl."
I modify the `config.gotmpl` to the code below and all tests pass:
```
{{- if .Var }}{{ formatName .Var }}{{- if .IsUnion }}{{ " : " }}{{- else }}{{ " = " }}{{- end }}{{- end }}{{- if .Name }}{{ formatName .Name }}{{- end }}{{- if .Data }}{{- "{\n" }}
{{- range .Data -}}
{{- indentLines (include "data"  .) "    " }}
{{- end -}}
{{- "}" }}{{- else }}{}{{- end }}
```
Did I miss anything? I would appreciate it if you could provide a little more detail about it. Thanks❤

Comment (Peefy) [reply pkg/tools/gen/templates/kcl/config.gotmpl]: You are right, I was wrong. I have modified the corresponding code.

**Review by jakezhu9 — APPROVED** [review]
Good Job! LGTM!

**Pull request merged**